src/install.c

```diff
@@ -152,25 +152,17 @@ gboolean determine_slot_states(GError **error)
 		}
 	}
 
-
-	if (booted) {
-		booted->state = ST_BOOTED;
-		g_debug("Found booted slot: %s on %s", booted->name, booted->device);
-	}
-
 	/* Determine active group members */
 	for (GList *l = slotlist; l != NULL; l = l->next) {
 		RaucSlot *s = (RaucSlot*) g_hash_table_lookup(r_context()->config->slots, l->data);
 
-		if (s->parent) {
-			if (s->parent->state & ST_ACTIVE) {
-				s->state |= ST_ACTIVE;
-			} else {
-				s->state |= ST_INACTIVE;
-			}
+		if (s == booted) {
+			s->state = ST_BOOTED;
+			g_debug("Found booted slot: %s on %s", s->name, s->device);
+		} else if (s->parent && s->parent == booted) {
+			s->state = ST_ACTIVE;
```

Comment:
Hmm, looking at this, I think it already failed if the parent itself has a parent. I that case we should probably fix all parent links to be direct, in the config parsing.

Reply:
Oh hmm that's interesting. So are you thinking, some code before this which will catch where a slot has a parent= which then itself has a parent=, and fix it so that all children in the group refer to the "real" parent?

Would it make more sense to just reject that configuration with an error, like these others?

Reply:
I suspect that there might be configs which have that setup. And the intent is clear, so we can just fix it at runtime (maybe print a warning when performing the fixup).

Reply:
Sure, makes sense. I will update the PR tomorrow.